pkg/controller/component/component_controller.go

```diff
@@ -127,27 +130,15 @@ func (r *ReconcileComponent) Reconcile(request reconcile.Request) (reconcile.Res
 			log.Error(err, "** Setting owner reference fails **")
 			return reconcile.Result{}, err
 		}
-
-		// Create an empty image name "nodejs-runtime"
-		// todo(corinne): check if we can reuse openshift's image stream
-		newImageForRuntime := newImageStreamFromDocker(instance.Namespace, instance.Name, instance.Spec.BuildType)
-		err = r.client.Create(context.TODO(), newImageForRuntime)
+		// Create a build image named either "myapp-builder" or reuse openshift "nodejs" builder image
+		ir, err := r.getBuilderImage(instance)
```

Comment:
An improvement to the code style would be to define 'builder image resolvers' and pass them elegantly - such that the first attempt would be to lookup in the `openshift` namespace and then the static list.

Reply:
Good idea, we can do it in a separate PR.
first let's add unit tests with this Pr
and then odc-132 to add e2e test (plus some RBAC correction needed)
Once we got a good test coverage, refactor the code for more elegant approach.

Reply:
OK, please create issues and track it. Add a comment in the code too.

Reply:
@sbose78 https://jira.coreos.com/browse/ODC-242 crested